src/style.rs

```diff
@@ -93,96 +92,91 @@ impl Style {
     /// Attempts to apply the given style to the given terminal. If
     /// the style is not supported, either there is no effect or else
     /// a similar, substitute style may be applied.
-    pub fn apply<T: Terminal + ?Sized>(self, term: &mut T) -> term::Result<()> {
-        term.reset()?;
+    pub fn apply<T: std::io::Write>(self, term: &mut T) -> std::io::Result<()> {
+        let mut s = anstyle::Style::new();
+        s.write_reset_to(term)?;
 
         macro_rules! fg_color {
             ($color:expr, $term_color:ident) => {
                 if self.contains($color) {
-                    if term.supports_color() {
```

Comment:
I think this functionality is lost in this PR. That is, if the thing we output to isn't a proper terminal (a file), we used to not add any styling, while we now probably now always output ANSI escape codes - which can probably be annoying when redirecting stderr to a log file.

The simplest approximation is probably to use a bound `std::io::Write + std::io::IsTerminal`, and use `is_terminal` as a replacement for `supports_color()` (it's probably not a proper replacement but I've seen `is_terminal` being used for that in practice).

Reply:
The approach you mention is also the official recommendation of the termcolor library, which seems like a relevant endorsement: https://docs.rs/termcolor/latest/termcolor/#detecting-presence-of-a-terminal

Reply:
Ah, fair point, I didn't know about `std::io::IsTerminal`. I think for Lalrpop, the functionality was maintained because there I started using AutoStream/StripStream which manage this for me. I pushed the change that has been suggested here. It does break my lalrpop implementation because `AutoStream`/`StripStream` don't implement `std::io::IsTerminal`. It might be related to `IsTerminal` being a sealed trait? Some trait shenanigans I think.

Maybe I don't need Anstream on the Lalrpop side if all of the style/color checking can happen here?